lib/localisation/src/preprocessing/keyence.rs

```diff
@@ -1 +1,54 @@
+use heapless::Vec;
 
+#[derive(PartialEq, Debug)]
+pub enum SensorChecks {
+    Acceptable,
+    Unnaceptable,
+}
+
+pub struct KeyenceAgrees {
+    keyence_data: Vec<bool, 2>,
+    previous_keyance_agreement: bool,
+}
+
+impl KeyenceAgrees {
+    pub fn new(keyence_data: Vec<bool, 2>) -> Self {
+        KeyenceAgrees {
+            keyence_data,
+            previous_keyance_agreement: true,
+        }
+    }
+
+    pub fn check_keyence_agrees(&mut self) -> SensorChecks {
```

Comment:
```suggestion
    pub fn check_keyence_agrees(&mut self, keyence_data: Vec<bool, 2>) -> SensorChecks {
```
The new keyence data should be passed into this function rather than `new` because once a new `KeyenceAgrees` has been created we'll just keep calling `check_keyence_agrees` with new values